js/components/common/Disclaimer.js

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+
+/**
+ * A React component representing the disclaimer popup
+ */
+export default class Disclaimer extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dontShowDisclaimer: localStorage.getItem('DontShowDisclaimer')
+        }
+    }
+
+    handleClose = () => {
+        let disclaimer = document.getElementById("disclaimerDiv");
+        disclaimer.classList.toggle("dismiss");
+    }
+
+    handleCheck = () => {
+        localStorage.setItem('DontShowDisclaimer', 'true');
+    }
+
+    render() {
+        const timetable = <a href="https://timetable.iit.artsci.utoronto.ca/">Official Timetable</a>;
+        const calendar = <a href="http://calendar.artsci.utoronto.ca/">Calendar</a>;
+
+        if (this.state.dontShowDisclaimer === 'true') {
+            return (null);
+        } else {
+            return (
+                <div id="disclaimerDiv" className="popup-banner">
```

Comment:
Since you've added the classes, I suspect that the ids here and for the inner input/button elements are no longer necessary, and you can remove them.